### PR TITLE
Fixing our build

### LINKED ISSUE #235: Fix api build
# Requirements

- [ ] running `yarn prod:build` doesn't work. There's several errors. Fix it!

### LINKED ISSUE #236: Create fake data
# Requirements

- [ ] We want to be able to load up our mock firebase db with fake test data. 
- [ ] We want to be able to run tests and use fake data to save/retrieve. 
- [ ] This should be based on the gql types

### LINKED ISSUE #237: Update Node version to latest and update packages
# Requirements

- [ ] Update node to v20+ and verify everything works
- [ ] Update all our package.json dependencies and verify them
- [ ] Upgrade yarn to latest version 
- [ ] Update our docker files to use the update version of node

### LINKED ISSUE #238: Fix missing types in gqlTypes.ts
# Requirements

- [ ] Fix the graphql typescript types code generator to properly output the types
- [ ] remove gqlTypes.ts from gitignore. (not really any reason we don't want to check it in. When it's updated, we can just check in the new changes)
- [ ] Add a pre task to auto run the generate so we are matching what prod build experiences.

### PR DESCRIPTION
**Pull request by ermish**
Resolves #235 
Resolves #236
Resolves #237
Resolves #238

#This PR fixes the current production build

# What changed
- Fixed the build which required many prerequisite fixes listed above. 
- This is a first step towards getting our CI/CD pipeline up and running! 
- The fake data and changes in services aren't perfect. It's just a starting point 😄  feel free to update
- updated yarn to the latest version. 

# Testing instructions
- run `yarn prod:build` and see no errors

### CODE REVIEW

**Comment by theyokohamalife** [comment]
@ermish  Can you separate this into 4 different PRs and make sure CI is passing for each?

When I run `yarn prod:build` it gives an error: 

```
YAMLException: bad indentation of a mapping entry at line 366, column 23:
        "@babel/highlight": ^7.22.5
                          ^
        "@babel/highlight": ^7.22.5
                          ^
```

**Review by theyokohamalife — CHANGES_REQUESTED** [review]
Off to a good start! ✨ Let's separate these into different PRs since they address very different issues. Also, let's make sure that the fake data is correct before merging. We have more active members in the repo now with various skill levels and we want to make sure that everything that gets merged into `main` is functional.

**Comment by theyokohamalife** [comment]
Also, you might want to rebase because it's showing 33 commits.

**Comment by theyokohamalife** [comment]
@ermish Don't forget to include descriptive PR titles.

File: src/fakeData/facilities.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            phone: '08000000000',
+            website: 'https://zoo.test.com',
+            mapsLink: ''
+        },
+        healthcareProfessionalIds: [],
+        healthcareProfessionals: []
+    }
+
+    return [facility]
+}
```
Comment (theyokohamalife) [inline src/fakeData/facilities.ts:26]: This file was removed in #227 because it wasn't being used. Is there a reason we need to reintroduce the fake data?

File: src/fakeData/healthcareProfessionals.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        names: [name],
+        degrees: [englishDegree],
+        spokenLanguages: [english],
+        specialties: [pandaSpecialty],
+        acceptedInsurance: [Insurance.InternationalHealthInsurance],
+        isDeleted: false
+    }
+
+    return [healthcareProfessionalOne, healthcareProfessionalTwo]
+}
```
Comment (theyokohamalife) [inline src/fakeData/healthcareProfessionals.ts:80]: This file was also removed in #227

File: src/fakeData/facilities.ts (hunk 1 above)
Comment (ermish) [reply src/fakeData/facilities.ts:26]: yes, we want fake data. 
1) this will be used for when we want to test endpoints by adding data, updating it, then deleting it.
2) we will also want seed data for when we load up a fake/test database so the site isn't empty. 

Keep in mind this is just a starting point and we'll need to update this iteratively.

**Comment by ermish** [comment]
> Also, you might want to rebase because it's showing 33 commits.

yup! I might put this in draft as it was working, but I rebased and have to fix it again. I'll ping you when it's actually ready to be tested.

File: src/databaseSeedTool.ts [hunk 3]
```diff
@@ -1,13 +1,12 @@
-import { initializeDb } from './database'
 import { addHealthcareProfessional } from './services/healthcareProfessionalService'
 import { addFacility } from './services/facilityService'
 
-import { fakeHealthcareProfessionals } from './fakeData/healthcareProfessional'
+import { fakeHealthcareProfessionals } from './fakeData/healthcareProfessionals'
```
[the remark this answers is not in the document]
Comment (ermish) [reply src/databaseSeedTool.ts:4]: for tests, yes, but we're likely going to want initial seed data to have the site functional beyond testing a single endpoint. So basically, we'll want both.

File: src/fakeData/healthcareProfessionals.ts (hunk 2 above)
Comment (theyokohamalife) [reply src/fakeData/healthcareProfessionals.ts:80]: @ermish I asked this in the issue, but I'll ask it again here. What can this fake data do that auto-mocking from the schema can't? It seems like extra work to manually seed the db with files that will have to be updated and maintained.

**Comment by ermish** [comment]
ok @theyokohamalife, build works again! please check

**Comment by ermish** [comment]
> > ok @theyokohamalife, build works again! please check
> 
> @ermish I get the following error after your latest commit:
> 
> Screenshot 2023-08-28 at 10 20 19 PM

hmmm I don't think you have the latest commit. It fixes that. 🤔

**Comment by theyokohamalife** [comment]
> hmmm I don't think you have the latest commit. It fixes that. 🤔

You're right. Just pulled again and it built without errors. 👍🏾

I think this is good to go after you get CI to pass.

File: src/databaseSeedTool.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ermish) [reply src/databaseSeedTool.ts:4]: that's what these are 😆

File: src/fakeData/healthcareProfessionals.ts (hunk 2 above)
Comment (ermish) [reply src/fakeData/healthcareProfessionals.ts:80]: answered in the previous comment:

> yes, we want fake data.
> this will be used for when we want to test endpoints by adding data, updating it, then deleting it.
> we will also want seed data for when we load up a fake/test database so the site isn't empty.
> Keep in mind this is just a starting point and we'll need to update this iteratively.

**Comment by ermish** [comment]
> > hmmm I don't think you have the latest commit. It fixes that. 🤔
> 
> @ermish You're right. Just pulled again and it built without errors. 👍🏾
> 
> I think this is good to go after you get CI to pass.

CI Passed! It's just the in progress workflows that are failing. I can comment them out I guess to have it pass

**Comment by ermish** [comment]
ok @theyokohamalife it's ready, please approve!

**Review by theyokohamalife — APPROVED** [review]
LGTM! Nice work!
